src/brushes/pencil_brush.class.js

```diff
@@ -246,7 +296,7 @@
       var zoom = this.canvas.getZoom(), adjustedDistance = Math.pow(distance / zoom, 2),
           i, l = points.length - 1, lastPoint = points[0], newPoints = [lastPoint],
           cDistance;
-      for (i = 1; i < l; i++) {
+      for (i = 1; i <= l; i++) {
```

Comment:
Needs to be validated against #6966

Reply:
It is why there's a merge conflict

Reply:
so master is at < l -1 to fix a tail issue with decimate.
Why did you add the equal in?
If you explain we can solve the conflict.
But to me is not clear why this has changed.

Reply:
The logic of straight line pops the last point and push the last pointer.
When mouse up fires and `drawStraightLine=true` it decimated the end and removes the last straight line part.
I don't know know/understand what decimate does so I don't want to propose anything but this can be mitigated by pushing another point on mouse up if `drawStraightLine=true` (I don't like it though)

Reply:
You can see/test
https://codesandbox.io/s/blue-shape-evc6t?file=/src/ShiftBrush.js:9550-9576